PlayerCollection.cs

```diff
@@ -33,21 +32,37 @@ public PlayerCollection(ConcurrentDictionary<string, CommonFields.Player> arcPla
 
         public void AddPlayer(CommonFields.Player arcPlayer, Func<uint, uint, AsyncTexture2D> iconGetter, ObservableCollection<string> availableRoles)
         {
-            if (_players.ContainsKey(arcPlayer.CharacterName))
+            Player previousCharacter = null;
+            if (_players.TryGetValue(arcPlayer.AccountName, out var existingPlayer))
             {
-                // Move from former players if player rejoined
                 var playerDisplay = GetPlayer(arcPlayer);
+
+                // Move from former players if player rejoined
                 if (playerDisplay?.IsFormerSquadMember ?? false)
                 {
                     playerDisplay.MoveFormerSquadMemberToActivePanel();
                 }
+                if (arcPlayer.CharacterName != existingPlayer.CharacterName)
+                {
+                    var newCharacter = new Player(arcPlayer, existingPlayer);
+                    playerDisplay.UpateCharacter(newCharacter);
+                    _players[arcPlayer.AccountName] = newCharacter;
+                }
                 //Don't add duplicate player
                 return;
+
             }
 
-            var player = new Player(arcPlayer.AccountName, isSelf: arcPlayer.Self, characterName: arcPlayer.CharacterName, profession: arcPlayer.Profession, currentSpecialization: arcPlayer.Elite);
+            var player = new Player(arcPlayer);
+            if (previousCharacter != null)
```

Comment:
Remark: by following this code, `previousCharacter` is always null because the code path where it gets a value affected finishes with a `return;`.

Reply:
`previousCharacter` is a relic from before I came up with the new constructor approach. Its value is never set anymore, I'll remove it.